src/components/PinPad/PinPad.tsx

```diff
@@ -0,0 +1,75 @@
+import { useRef, useState, useEffect } from 'react';
+import ReactNativePinView from "react-native-pin-view";
+import BottomDrawer, {
+  BottomDrawerMethods,
+} from 'react-native-animated-bottom-drawer';
+import { Icon } from "@components";
+import { faDeleteLeft } from "@fortawesome/free-solid-svg-icons";
+import * as S from "./styled";
+
+type PinViewFunctions = {
+  clearAll: () => void;
+};
+
+type PinPadProps = {
+  onPinEntered: (pin: string) => void;
+}
+
+export const PinPad = (props: PinPadProps) => {
+  const pinView = useRef<PinViewFunctions>(null);
+  const bottomDrawerRef = useRef<BottomDrawerMethods>(null);
+  const [showRemoveButton, setShowRemoveButton] = useState(false);
+  const [enteredPin, setEnteredPin] = useState("");
+  const [buttonPressed, setButtonPressed] = useState("");
+
+  useEffect(() => {
+    if (enteredPin.length > 0) {
+      setShowRemoveButton(true)
+    } else {
+      setShowRemoveButton(false)
+    }
+    if (enteredPin.length === 4) {
+        props.onPinEntered(enteredPin)
+    }
+  }, [enteredPin]);
+
+  useEffect(() => {
+      if (buttonPressed === "custom_left" && pinView.current) {
+        pinView.current.clearAll()
+      }
+  }, [buttonPressed])
+
+  return (
+    <>
+        <BottomDrawer
+            ref={bottomDrawerRef}
+            openOnMount gestureMode={"none"}
+            closeOnBackdropPress={false}
+            initialHeight={535}
+            backdropOpacity={0}
+            customStyles={S.BottomDrawerStyles}
+        >
+            <S.PinPadContainer>
+              <S.PinPadTitle>
+                Boltcard PIN
+              </S.PinPadTitle>
+              <ReactNativePinView
+                inputSize={32}
+                ref={pinView}
```

Comment:
To avoid a new TS issue here, add a `// @ts-ignore` :

```suggestion
                // @ts-ignore
                ref={pinView}
```